### PR TITLE
add gosdk interface

### PR DESCRIPTION
**Pull request by rizary**
A brief description of the changes in this PR:
change into gosdk interface.

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- gosdk: https://github.com/0chain/gosdk/pull/372

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/gosdk-interface  |
| zboxcli  | feature/gosdk-interface  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feature/gosdk-interface  |
| zboxcli  | feature/gosdk-interface  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: cmd/root.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ var bSilent bool
 var allocUnderRepair bool
 
 var walletJSON string
+var storageSdk sdk.StorageSdkSchema
```
Comment (cnlangzi) [inline cmd/root.go:35]: I will suggest to rename `storageSdk` with `sdk`, and import `sdk` as storageSdk. so all changes can be removed on all commands.

**Review by cnlangzi — CHANGES_REQUESTED** [review]
try it?

File: cmd/root.go (hunk 1 above)
Comment (rizary) [reply cmd/root.go:35]: Noted. I'll change that.

Comment (rizary) [reply cmd/root.go:35]: It'll create conflict with another sdk import.

**Comment by cnlangzi** [comment]
it is unnecessary and too many conflicts for now. let's reopen it if it is necessary in the future.

**Pull request closed without merging**